src/frontend/js/components/PaymentButton/components/PaymentInterfaces/LyraPopIn.tsx

```diff
@@ -0,0 +1,121 @@
+import KRGlue from '@lyracom/embedded-form-glue';
+import { useIntl } from 'react-intl';
+import { useRef } from 'react';
+import * as Joanie from 'types/Joanie';
+import { PaymentErrorMessageId } from 'components/PaymentButton/index';
+import { useAsyncEffect } from 'hooks/useAsyncEffect';
+import { handle } from 'utils/errors/handle';
+
+const LYRA_FORM_ID = 'lyra-form';
+
+const LyraPopIn = ({
+  onSuccess,
+  onError,
+  configuration,
+  form_token,
+}: Joanie.PaymentInterfaceProps<Joanie.LyraPayment>) => {
+  const intl = useIntl();
+  const shouldAbort = useRef<Boolean>(true);
+
+  const handleError = (error?: Error) => {
+    if (error) handle(`[LyraPopIn] - ${error}`);
+
+    if (shouldAbort) {
+      onError(PaymentErrorMessageId.ERROR_ABORTING);
+    }
+    onError(PaymentErrorMessageId.ERROR_DEFAULT);
+  };
+
+  const initializeLyraForm = async () => {
+    const { base_url: baseUrl, public_key: publicKey } = configuration;
+
+    const link = document.createElement('link');
+    link.setAttribute('rel', 'stylesheet');
+    link.setAttribute('href', `${baseUrl}/static/js/krypton-client/V4.0/ext/neon-reset.min.css`);
+    document.head.appendChild(link);
+
+    const script = document.createElement('script');
+    script.src = `${baseUrl}/static/js/krypton-client/V4.0/ext/neon.js`;
+    document.head.appendChild(script);
+
+    const { KR } = await KRGlue.loadLibrary(baseUrl, publicKey).catch((error) => {
+      handleError(error);
+      return { KR: null };
+    });
+
+    return KR;
+  };
+
+  useAsyncEffect(async () => {
+    const KR = window.KR ?? (await initializeLyraForm());
+    if (KR === null) return;
+
+    await KR.setFormConfig({
+      formToken: form_token,
+      'kr-language': intl.locale,
+      'kr-spa-mode': true,
+      'kr-z-index': '400',
+    });
+
+    // Create a container for the lyra form which must be out of the React tree.
+    // This is necessary to avoid the form being unmounted when the component is.
+    const formContainer = document.createElement('div');
+    formContainer.id = LYRA_FORM_ID;
+    formContainer.classList.add('kr-embedded');
+    formContainer.setAttribute('kr-popin', '');
+    formContainer.style.visibility = 'hidden';
+    document.body.appendChild(formContainer);
+
+    const {
+      result: { formId },
+    } = await KR.renderElements('#lyra-form');
+
+    const handleFormSubmit = async (response: KRPaymentResponse) => {
+      shouldAbort.current = false;
+      await KR.closePopin(formId);
+      if (response.clientAnswer.orderStatus === 'PAID') {
+        onSuccess();
+        return true;
+      }
+
+      handleError();
+      return false;
+    };
+
+    const handleFormError = async (error: KRError) => {
+      // Do not close the pop-in if the error is a client error.
```

Comment:
curious: why? 👀

Reply:
Because the form in the pop display explicit error message and user is able to update what was wrong and resubmit the form.

Reply:
I imagined that client error are form error displayed in the popin.